python/Dockerfile

```diff
@@ -1,16 +1,16 @@
 # This list must match the versions specified in
 # python/lib/dependabot/python/language_version_manager.rb: PRE_INSTALLED_PYTHON_VERSIONS
-ARG PY_3_11=3.11.5
+ARG PY_3_12=3.12.1
+ARG PY_3_11=3.11.7
 ARG PY_3_10=3.10.13
 ARG PY_3_9=3.9.18
-ARG PY_3_8=3.8.18
-ARG PYENV_VERSION=v2.3.25
+ARG PYENV_VERSION=v2.3.35
 
 FROM ghcr.io/dependabot/dependabot-updater-core as python-core
+ARG PY_3_12
 ARG PY_3_11
 ARG PY_3_10
 ARG PY_3_9
-ARG PY_3_8
```

Comment:
🤔 We published [a changelog post](https://github.blog/changelog/2023-08-02-deprecation-notice-dependabot-to-drop-support-for-python-3-6-and-3-7/) for dropping 3.6/3.7 support. What do you think about dropping support in another PR, perhaps with a similar approach (or publishing some kind of signal that states we'll continually drop support for outdated versions)?

Reply:
We're not dropping support for it, it's just not installed by default so when running a job that needs this version, we'll first install it

Reply:
Oh looks like we're not installing stuff on the fly anymore, I'll put it back